x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/related_integrations/related_integrations.test.tsx

```diff
@@ -0,0 +1,698 @@
+
+      expect(visibleIntegrations[0]).toHaveTextContent('Package-a');
+      expect(visibleVersionInputs[0]).toHaveValue('1.2.3');
+
+      expect(visibleIntegrations[1]).toHaveTextContent('Package-b integration-a');
+      expect(visibleVersionInputs[1]).toHaveValue('3.2.1');
+    });
+  });
+
+  describe('valid form submitting', () => {
+    it('returns undefined when no integrations are selected', async () => {
+      const handleSubmit = jest.fn();
+
+      render(<TestForm onSubmit={handleSubmit} />, { wrapper: createReactQueryWrapper() });
+
+      await submitForm();
+      await waitFor(() => {
+        expect(handleSubmit).toHaveBeenCalled();
+      });
+
+      expect(handleSubmit).toHaveBeenCalledWith({
+        data: undefined,
+        isValid: true,
+      });
+    });
+
+    it('returns a selected integration', async () => {
+      fleetIntegrationsApi.fetchAllIntegrations.mockResolvedValue({
+        integrations: [
+          {
+            package_name: 'package-a',
+            package_title: 'Package A',
+            latest_package_version: '1.2.0',
+            is_installed: false,
+            is_enabled: false,
+          },
+        ],
+      });
+
+      const handleSubmit = jest.fn();
+
+      render(<TestForm onSubmit={handleSubmit} />, { wrapper: createReactQueryWrapper() });
+
+      await addRelatedIntegrationRow();
+      await selectEuiComboBoxOption({
+        comboBoxToggleButton: screen.getByTestId(COMBO_BOX_TOGGLE_BUTTON_TEST_ID),
+        optionIndex: 0,
+      });
+      await submitForm();
+
+      expect(handleSubmit).toHaveBeenCalledWith({
+        data: [{ integration: undefined, package: 'package-a', version: '^1.2.0' }],
+        isValid: true,
+      });
+    });
+
+    it('returns a selected integration with version constraint modified ', async () => {
+      fleetIntegrationsApi.fetchAllIntegrations.mockResolvedValue({
+        integrations: [
+          {
+            package_name: 'package-a',
+            package_title: 'Package A',
+            latest_package_version: '1.2.0',
+            is_installed: false,
+            is_enabled: false,
+          },
+        ],
+      });
+
+      const handleSubmit = jest.fn();
+
+      render(<TestForm onSubmit={handleSubmit} />, { wrapper: createReactQueryWrapper() });
+
+      await addRelatedIntegrationRow();
+      await selectEuiComboBoxOption({
+        comboBoxToggleButton: screen.getByTestId(COMBO_BOX_TOGGLE_BUTTON_TEST_ID),
+        optionIndex: 0,
+      });
+      await setVersion({ input: screen.getByTestId(VERSION_INPUT_TEST_ID), value: '1.0.0' });
+      await submitForm();
+
+      expect(handleSubmit).toHaveBeenCalledWith({
+        data: [{ integration: undefined, package: 'package-a', version: '1.0.0' }],
+        isValid: true,
+      });
+    });
+
+    it('returns a saved earlier integration', async () => {
+      fleetIntegrationsApi.fetchAllIntegrations.mockResolvedValue({
+        integrations: [
```

Comment:
Perhaps we can also add an item where "integration" is not undefined to test that it gets submitted too